lib/shared/layers/python-sdk/python/genai_core/embeddings.py

```diff
@@ -73,30 +73,47 @@ def _generate_embeddings_openai(
 
 
 def _generate_embeddings_bedrock(
-    model: genai_core.types.EmbeddingsModel, input: List[str]
+    model: genai_core.types.EmbeddingsModel, input: List[str], task: str = "store"
 ):
     bedrock = genai_core.clients.get_bedrock_client()
 
     if not bedrock:
         raise genai_core.types.CommonError("Bedrock is not enabled.")
 
     ret_value = []
-    for value in input:
-        body = json.dumps({"inputText": value})
+    model_provider = model.name.split(".")[0]
+    if model_provider == "amazon":
```

Comment:
2 suggestions:
- Let's have providers in [types.py](https://github.com/aws-samples/aws-genai-llm-chatbot/blob/main/lib/shared/layers/python-sdk/python/genai_core/types.py) instead of hardcoded at each condition. 
- better to break down the provider-related code into separate:
  - `_generate_embeddings_amazon`
  - `_generate_embeddings_cohere` or any better naming idea 
  and just have the condition statement in this main `_generate_embeddings_bedrock`